tests/ops/service/connectors/saas/test_authenticated_client.py

```diff
@@ -0,0 +1,121 @@
+import time
+import unittest.mock as mock
+from email.utils import formatdate
+from typing import Any, Dict
+
+import pytest
+from requests import ConnectionError, Response, Session
+
+from fidesops.ops.common_exceptions import ClientUnsuccessfulException
+from fidesops.ops.models.connectionconfig import ConnectionConfig, ConnectionType
+from fidesops.ops.schemas.saas.shared_schemas import HTTPMethod, SaaSRequestParams
+from fidesops.ops.service.connectors.saas.authenticated_client import (
+    AuthenticatedClient,
+    get_retry_after,
+)
+from fidesops.ops.util.saas_util import load_config_with_replacement
+
+
+@pytest.fixture
+def test_saas_config() -> Dict[str, Any]:
+    return load_config_with_replacement(
+        "data/saas/config/segment_config.yml",
+        "<instance_fides_key>",
+        "test_config",
+    )
+
+
+@pytest.fixture
+def test_connection_config(test_saas_config) -> ConnectionConfig:
+    return ConnectionConfig(
+        key="test_config",
+        connection_type=ConnectionType.saas,
+        saas_config=test_saas_config,
+        secrets={"access_token": "test_token"},
+    )
+
+
+@pytest.fixture
+def test_saas_request() -> SaaSRequestParams:
+    return SaaSRequestParams(
+        method=HTTPMethod.GET,
+        path="test_path",
+        query_params={},
+    )
+
+
+@pytest.fixture
+def test_authenticated_client(test_connection_config) -> AuthenticatedClient:
+    return AuthenticatedClient("https://test_uri", test_connection_config)
+
+
+@pytest.mark.unit_saas
+@mock.patch.object(Session, "send")
+class TestAuthenticatedClient:
```

Comment:
the mocking here is nicely done - these tests are really easy to follow.

Reply:
ty ty, surprised it ended up working so nicely tbh